src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -248,6 +248,11 @@ internal void ValidateOptions(DiagnosticBag diagnostics, CommonMessageProvider m
             {
                 diagnostics.Add(messageProvider.CreateDiagnostic(messageProvider.ERR_InvalidSubsystemVersion, Location.None, SubsystemVersion.ToString()));
             }
+
+            if (!IncludePrivateMembers && !EmitMetadataOnly)
```

Comment:
>if (!IncludePrivateMembers && !EmitMetadataOnly) [](start = 12, length = 48)

As we discussed offline, this probably wouldn't work for the case when we are emitting a real assembly and a ref assembly together. Also, this could be a possible breaking change for existing code.